[DO NOT MERGE] App strings reworked

comradekingu opened the pull request:
Would be great to have this available for translation on Weblate in some form.

comradekingu committed:
App strings reworked

nickcalyx commented:
https://hosted.weblate.org/projects/calyxos/seedvault/

chirayudesai commented:
There are a lot of good changes here, but there's also some changes that I don't agree with. I only had a very cursory look and will come back to it again, but we had the same issue last time - this is such a big change all at once, making it harder to get merged.


I understand that it's easier to do it all at once, but smaller changes are easier for us to review and merge (like say anything that doesn't reword the string entirely, just changes application -> app and things like that)

I'll wait to see what the others have to say about it.

weblate committed:
Added translation using Weblate (Asturian)

Added translation using Weblate (Gaelic)

Added translation using Weblate (Asturian)

Added translation using Weblate (Welsh)

Added translation using Weblate (Gaelic)

Added translation using Weblate (Welsh)

Added translation using Weblate (Asturian)

Added translation using Weblate (Gaelic)

Added translation using Weblate (Welsh)

Co-authored-by: Weblate <noreply@weblate.org>

Commit:
Translated using Weblate (Japanese)

Currently translated at 93.7% (15 of 16 strings)

Translated using Weblate (Japanese)

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (Japanese)

Currently translated at 88.1% (149 of 169 strings)

Co-authored-by: lx3k <lx3kvdms@anonaddy.me>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ja/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/ja/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ja/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 87.5% (14 of 16 strings)

Translated using Weblate (Norwegian Bokmål)

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (Norwegian Bokmål)

Currently translated at 76.9% (130 of 169 strings)

Co-authored-by: Allan Nordhøy <epost@anotheragency.no>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/nb_NO/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

josprachi committed:
Translated using Weblate (Marathi)

Currently translated at 100.0% (1 of 1 strings)

Co-authored-by: Prachi Joshi <josprachi@yahoo.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/mr/
Translation: CalyxOS/Seedvault contacts backup

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Hebrew)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Hebrew)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Yaron Shahrabani <sh.yaron@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/he/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/he/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

oersen committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Turkish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Turkish)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/tr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/tr/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (German)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (German)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (German)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: nautilusx <translate@disroot.org>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/de/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/de/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Óscar Fernández Díaz <oscfdezdz@tuta.io>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/es/
Translation: CalyxOS/Seedvault

Commit:
Translated using Weblate (Arabic)

Currently translated at 100.0% (16 of 16 strings)

Translated using Weblate (Arabic)

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (Arabic)

Currently translated at 13.5% (23 of 170 strings)

Co-authored-by: fawaz006 <fawaz27000@hotmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ar/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/ar/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ar/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Eric <alchemillatruth@purelymail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/zh_Hans/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (Italian)

Currently translated at 70.0% (119 of 170 strings)

Co-authored-by: Pietro Cappuccino <p.cappuccino@tiscali.it>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/it/
Translation: CalyxOS/Seedvault

quantien51 committed:
Translated using Weblate (Vietnamese)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Doan Quan Tien <doanquantien1@outlook.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/vi/
Translation: CalyxOS/Seedvault

mikeNG committed:
Translated using Weblate (Greek)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Greek)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Greek)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Michael Bestas <mkbestas@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/el/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/el/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

Micovec committed:
Translated using Weblate (Czech)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Czech)

Currently translated at 97.2% (174 of 179 strings)

Translated using Weblate (Czech)

Currently translated at 96.6% (173 of 179 strings)

Translated using Weblate (Czech)

Currently translated at 99.4% (169 of 170 strings)

Co-authored-by: Micovec <jiri.vokrinek@email.cz>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/cs/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/cs/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

Lisapple committed:
Translated using Weblate (French)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (French)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Maxime Leroy <lisacintosh@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/fr/
Translation: CalyxOS/Seedvault

Commit:
Translated using Weblate (Catalan)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Catalan)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Catalan)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: kbci_wcn <kbci_wcn@aleeas.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ca/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ca/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

creepen123 committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Russian)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Russian)

Currently translated at 98.8% (177 of 179 strings)

Translated using Weblate (Russian)

Currently translated at 100.0% (170 of 170 strings)

Co-authored-by: Егор Ермаков <eg.ermakov2016@yandex.ru>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ru/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ru/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

unbiaseduser-github committed:
Translated using Weblate (Vietnamese)

Currently translated at 100.0% (179 of 179 strings)

Co-authored-by: bruh <quangtrung02hn16@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/vi/
Translation: CalyxOS/Seedvault

Launacloud committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 98.3% (176 of 179 strings)

Co-authored-by: Launacloud <CLOUDAPPSLAURA@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pt_BR/
Translation: CalyxOS/Seedvault

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Co-authored-by: JBL <jb.lze@tutanota.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/fr/
Translation: CalyxOS/Seedvault storage backup

NorwayFun committed:
Translated using Weblate (Georgian)

Currently translated at 100.0% (1 of 1 strings)

Co-authored-by: Temuri Doghonadze <temuri.doghonadze@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/ka/
Translation: CalyxOS/Seedvault contacts backup

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Galician)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Galician)

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (179 of 179 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/gl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/gl/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (179 of 179 strings)

Co-authored-by: Vri 🌈 <weblate@vrifox.cc>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/de/
Translation: CalyxOS/Seedvault

Mart1997 committed:
Translated using Weblate (Spanish (American))

Currently translated at 94.1% (16 of 17 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Spanish (American))

Currently translated at 100.0% (1 of 1 strings)

Co-authored-by: Martin <martinjason12345678@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/es_US/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es_US/
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

SantosSi committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 88.2% (158 of 179 strings)

Co-authored-by: ssantos <ssantos@web.de>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pt_PT/
Translation: CalyxOS/Seedvault

MarongHappy committed:
Translated using Weblate (Korean)

Currently translated at 100.0% (179 of 179 strings)

Translated using Weblate (Korean)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Korean)

Currently translated at 100.0% (179 of 179 strings)

Co-authored-by: 이정희 <daemul72@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ko/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ko/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault storage backup

70h committed:
Translated using Weblate (Lithuanian)

Currently translated at 100.0% (17 of 17 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (179 of 179 strings)

Co-authored-by: Gediminas Murauskas <muziejusinfo@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/lt/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/lt/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/lt/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

Commit:
Translated using Weblate (French (Canada))

Currently translated at 100.0% (1 of 1 strings)

Translated using Weblate (French (Canada))

Currently translated at 4.4% (8 of 179 strings)

Co-authored-by: blueline <blueline3@proton.me>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/fr_CA/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/fr_CA/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup

chirayudesai commented:
Friendly bump :)

@mikeNG perhaps we can manually extract and use some of these?

chirayudesai committed:
Revert "Translations update from Hosted Weblate" (#467)

Accidentally went into android12

comradekingu committed:
Merge branch 'android12' into patch-2

comradekingu commented:
@chirayudesai
Serves as yet another example of how to wrestle inconsistency from the hands of what has already been fixed.
Every single translation is out of date for a start.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="backup">Backup</string>
     <string name="data_management_label">Seedvault Backup</string>
-    <string name="current_destination_string">Backup status and settings</string>
+    <string name="current_destination_string">Back up status and settings</string>
```
```suggestion
    <string name="current_destination_string">Status and settings for backups</string>
```
If not this, "backup" is a noun that is out of place. Translators are left guessing (as I did), and subsequently users for what to make of the translation.

chirayudesai replied:
"Backup" as a noun seems like the correct usage here.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
 
     <!-- Settings -->
     <string name="settings_category_apps">App backup</string>
-    <string name="settings_backup">Backup my apps</string>
```
This is for sure wrong, which is my confidence level for changing other such instances.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_backup_last_backup_never">Never</string>
-    <string name="settings_info">All backups are encrypted on your phone. To restore from backup you will need your 12-word recovery code.</string>
-    <string name="settings_auto_restore_title">Automatic restore</string>
+    <string name="settings_info">All backups are encrypted on your device. Restore backups using your 12-word recovery code.</string>
```
The user isn't necessarily on a phone.
Note that it is "recovery code" here.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_auto_restore_summary">When reinstalling an app, restore backed up settings and data.</string>
     <string name="settings_auto_restore_summary_usb">Note: Your %1$s needs to be plugged in for this to work.</string>
-    <string name="settings_category_app_data_backup">App data backup</string>
+    <string name="settings_category_app_data_backup">App-data backup</string>
```
```suggestion
    <string name="settings_category_app_data_backup">Backup of app data</string>
```
Can't run three nouns in sequence.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_backup_location_internal">Internal storage</string>
     <string name="settings_backup_last_backup_never">Never</string>
-    <string name="settings_info">All backups are encrypted on your phone. To restore from backup you will need your 12-word recovery code.</string>
-    <string name="settings_auto_restore_title">Automatic restore</string>
```
Automatic restoration, or automatic restoring would be possible options, this is however wrong.

chirayudesai replied:
Restoration could work.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_backup_location_none">None</string>
     <string name="settings_backup_location_internal">Internal storage</string>
     <string name="settings_backup_last_backup_never">Never</string>
-    <string name="settings_info">All backups are encrypted on your phone. To restore from backup you will need your 12-word recovery code.</string>
```
The user isn't necessarily on a phone.
Note that it is "recovery code" here.

chirayudesai replied:
Yes. This could be device, or Android has a way to make it phone / tablet.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
+    <string name="settings_category_app_data_backup">App-data backup</string>
     <string name="settings_backup_apk_title">App backup</string>
-    <string name="settings_backup_apk_summary">Back up the apps themselves. Otherwise, only app data would get backed up.</string>
-    <string name="settings_backup_apk_dialog_title">Really disable app backup?</string>
```
The reason to not use "enable/disable" is it offers only higher entropy over "turn on/off".
A legitimate use would be if something can be on, while also disabled at some lower level, when this is a meaningful distinction to communicate to the user. I am not seeing it.

chirayudesai replied:
"Turn on/off" would make sense here.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_backup_apk_title">App backup</string>
-    <string name="settings_backup_apk_summary">Back up the apps themselves. Otherwise, only app data would get backed up.</string>
-    <string name="settings_backup_apk_dialog_title">Really disable app backup?</string>
-    <string name="settings_backup_apk_dialog_message">Disabled app backup will still back up app data. However, it will not get restored automatically.\n\nYou will need to install all your apps manually while having \"Automatic Restore\" switched on.</string>
```
We could argue over if this means what you think it means, but it sure doesn't do a good job of conveying meaning.

chirayudesai replied:
Absolutely agreed.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
     <string name="settings_backup_storage_battery_optimization">Warning: No automatic backups, because battery optimization is active.</string>
     <string name="settings_backup_new_code_dialog_title">New recovery code required</string>
     <string name="settings_backup_new_code_dialog_message">To continue using app backups, you need to generate a new recovery code.\n\nWe are sorry for the inconvenience.</string>
-    <string name="settings_backup_new_code_code_dialog_ok">New code</string>
```
And now it is just "code", whereas before it was "recovery code".
Is this a size/length restriction?

chirayudesai replied:
It is for a dialog so possibly, can check.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
     <string name="notification_error_action">Fix</string>
 
-    <string name="notification_restore_error_channel_title">Auto restore flash drive error</string>
-    <string name="notification_restore_error_title">Could not restore data for %1$s</string>
```
This error is "Could not".

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +177,26 @@
-    <string name="about_summary">A backup application using Android\'s internal backup API.</string>
+    <string name="about_summary">A backup app using Android\'s internal backup API.</string>
     <string name="about_version">Version: %s</string>
     <string name="about_license">License: <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache2</a></string>
```
```suggestion
    <string name="about_license">Licensed <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache 2.0</a></string>
```

chirayudesai replied:
`License:` is pretty standard, no?

`Apache 2.0` would be correct.

comradekingu replied:
I would go "Libre software, licensed Apache 2.0", and then add "use, study, change and share freely", postponed mentally into 2026 :)

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +177,26 @@
+    <string name="about_summary">A backup app using Android\'s internal backup API.</string>
     <string name="about_version">Version: %s</string>
     <string name="about_license">License: <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache2</a></string>
     <string name="about_author">Written by: <a href="https://github.com/stevesoltys">Steve Soltys</a> and <a href="https://blog.grobox.de">Torsten Grote</a></string>
```
```suggestion
    <string name="about_author">By <a href="https://github.com/stevesoltys">Steve Soltys</a> and <a href="https://blog.grobox.de">Torsten Grote</a></string>
```

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +177,26 @@
     <string name="restore_restoring_error_title">Unable to restore some apps</string>
-    <string name="restore_restoring_error_message">You can re-install these apps manually and "Automatic restore" will attempt to restore their data (when enabled).</string>
-    <string name="restore_magic_package">System package manager</string>
-    <string name="restore_finished_success">Restore complete</string>
```
I read this as "restore (something) in a complete fashion (?)"

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +177,26 @@
-    <string name="restore_finished_error">An error occurred while restoring the backup.</string>
+    <string name="restore_restoring_error_message">You can re-install these apps manually and "Auto-restore" will attempt to restore their data (when on).</string>
+    <string name="restore_magic_package">System package-manager</string>
+    <string name="restore_finished_success">Restoration complete</string>
```
```suggestion
    <string name="restore_finished_success">Restored</string>
```
More similar to "Backed up" (success) in https://github.com/seedvault-app/seedvault/pull/404/files#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103R148

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
 
     <string name="notification_error_channel_title">Error notification</string>
     <string name="notification_error_title">Backup error</string>
     <string name="notification_error_text">A device backup failed to run.</string>
```
```suggestion
    <string name="notification_error_text">Could not back up the device.</string>
```
"A backup run", (as jobs running) is a very hard concept to get across.

I am not so sure it is the entire device, which makes this cryptic.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
 
     <string name="notification_success_title">Backup finished</string>
     <string name="notification_success_text">%1$d of %2$d apps backed up. Tap to learn more.</string>
-    <string name="notification_failed_title">Backup failed</string>
```
There is no _text_ for this (?). Moreover, errors stopped being "Could not".

chirayudesai replied:
Failed does have a specific meaning in software though, and this fits.

comradekingu replied:
I would argue we aren't in software here though, and I don't know what this means.
I guess it means the process of backing up failed, rather than the backup of its own accord (due to for example bit-rot.)

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -138,19 +138,19 @@
     <string name="backup_app_in_progress">Backing up</string>
     <string name="backup_app_success">Backed up</string>
     <string name="backup_app_warning">Backup warning</string>
     <string name="backup_app_was_stopped">Not backed up as it wasn\'t used recently</string>
```
```suggestion
    <string name="backup_app_was_stopped">No backup possible</string>
```
(Changed after suggestion below)

chirayudesai replied:
This is the string I hate the most, and one that confuses most people.

In this case, the system refuses to backup the application, because it's not running due to some reason, or because it may not have data to backup.

"No backup possible" or something along those lines might be better, but they're all confusing.

comradekingu replied:
I think that should be split into two cases,
one for errors that are internal to the Seedvault app,
one for errors to do with apps having nothing to back up, or actively/effectively preventing it.

"Not possible" isn't always the case with the former case.

chirayudesai commented:
> Every single translation is out of date for a start.

So this'd have to be rebased on top of `android13`, as that's what we're using now.

Getting translations switched over, that should help.

I can rebase it for you (if you don't mind, and have it set to allow that) - but the other point about there being too many changes to merge at once still stands.

Some can be merged right away, some after a quick discussion, and then others pending further discussion.

Will start replying to comments in a bit.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -138,19 +138,19 @@
+    <!-- This text gets shown for apps that the OS did not try to back up for whatever reason e.g. no backup was run yet -->
     <string name="backup_app_not_yet_backed_up">Waiting to back up…</string>
     <string name="restore_app_not_yet_backed_up">Was not yet backed up</string>
     <string name="backup_app_in_progress">Backing up</string>
```
```suggestion
    <string name="backup_app_in_progress">Backing up…</string>
```
Missed this one.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +177,26 @@
     <string name="about_title">About</string>
-    <string name="about_summary">A backup application using Android\'s internal backup API.</string>
+    <string name="about_summary">A backup app using Android\'s internal backup API.</string>
     <string name="about_version">Version: %s</string>
```
```suggestion
    <string name="about_version">Version %s</string>
```

chirayudesai replied:
Why?

comradekingu replied:
Adding the colons is more staccato. It helps make it sentences in the case of "licensed", and to omit it where not needed (here). More natural and it saves characters.
Moreover it may help translators not make it a table-like structure, particularly if the verb-object-subject sequence is different in their languages.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
+    <string name="recovery_code_error_checksum_word">Your code is invalid. Please check all the words as well as their position and try again.</string>
     <string name="recovery_code_verification_ok_title">Recovery code verified</string>
-    <string name="recovery_code_verification_ok_message">Your code is correct and will work for restoring your backup.</string>
+    <string name="recovery_code_verification_ok_message">Your code is correct, so it can restore your backup.</string>
```
```suggestion
    <string name="recovery_code_verification_ok_message">Your recovery code is correct (so it can restore your backup).</string>
```

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
     <string name="recovery_code_verification_error_title">Incorrect recovery code</string>
     <string name="recovery_code_verification_error_message">You have entered an invalid recovery code. Please try again!\n\nIf you have lost your code, tap on \'Generate new code\' below.</string>
     <string name="recovery_code_verification_try_again">Try again</string>
     <string name="recovery_code_verification_generate_new">Generate new code</string>
```
```suggestion
    <string name="recovery_code_verification_generate_new">Generate new recovery code</string>
```

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
+    <string name="recovery_code_verification_new_dialog_message">Do you want to generating a new code, making your existing backups inaccessible? They will be deleted if possible.</string>
+    <string name="recovery_code_recreated">New recovery code created</string>
     <string name="recovery_code_auth_title">Re-enter your screen lock</string>
     <string name="recovery_code_auth_description">Enter your device credentials to continue</string>
```
```suggestion
    <string name="recovery_code_auth_description">Unlock your device to continue</string>
```
?

chirayudesai replied:
This is directly lifted from other places in the Android system I believe.

The device would already be unlocked at this point, it wants a confirmation of the unlock pin/pattern/password, hence the word device credentials.

comradekingu replied:
Fairly sure that runs afoul of the new language guidelines.
Not that it doesn't change very often, but might be a good idea to check if parity is the goal.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,45 +3,45 @@
-    <string name="settings_backup_apk_dialog_title">Really disable app backup?</string>
-    <string name="settings_backup_apk_dialog_message">Disabled app backup will still back up app data. However, it will not get restored automatically.\n\nYou will need to install all your apps manually while having \"Automatic Restore\" switched on.</string>
+    <string name="settings_backup_apk_summary">Also back up the apps themselves. Otherwise, only app data is backed up.</string>
+    <string name="settings_backup_apk_dialog_title">Really turn off app backup?</string>
```
```suggestion
    <string name="settings_backup_apk_dialog_title">Really turn off app backups?</string>
```
Is it just the app itself, or (as this would imply) all apps?

chirayudesai replied:
This turns off apk backup for all apps. Data would still get backed up.

comradekingu committed:
New strings

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +180,26 @@
-    <string name="restore_storage_in_progress_info">Your files are being restored in the background. You can start using your phone while this is running.\n\nSome apps (e.g. Signal or WhatsApp) might require files to be fully restored to import a backup. Try to avoid starting those apps before file restore is complete.</string>
+    <string name="restore_storage_in_progress_title">Restoring files…</string>
+    <string name="restore_storage_in_progress_info">You can use your device while this is running in the background.\n\nSome apps (e.g. Signal or WhatsApp) might require files to be fully restored to import a backup. Try to avoid starting those apps before file restoration completes.</string>
     <string name="restore_storage_got_it">Got it</string>
```
```suggestion
    <string name="restore_storage_got_it">OK</string>
```

chirayudesai commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
     <string name="recovery_code_verification_new_dialog_title">Wait one second…</string>
-    <string name="recovery_code_verification_new_dialog_message">Generating a new code will make your existing backups inaccessible. We\'ll try to delete them if possible.\n\nAre you sure you want to do this?</string>
-    <string name="recovery_code_recreated">New recovery code has been created successfully</string>
+    <string name="recovery_code_verification_new_dialog_message">Do you want to generating a new code, making your existing backups inaccessible? They will be deleted if possible.</string>
```
Do you want to **generate** ...

comradekingu replied:
I made  a new suggestion below.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,50 +3,53 @@
+    <string name="settings_expert_logcat_error">Could not save app log</string>
+    
     <!-- Storage Location -->
     <string name="storage_fragment_backup_title">Choose where to store backups</string>
```
```suggestion
    <string name="storage_fragment_backup_title">Choose where to save backups</string>
```
Perhaps less elegant, but "store" isn't different from "save" when it could be.
(Save to one place, store in another.)
"Storage location" and "backup location" used elsewhere.

chirayudesai commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +94,40 @@
     <string name="notification_success_title">Backup finished</string>
     <string name="notification_success_text">%1$d of %2$d apps backed up. Tap to learn more.</string>
-    <string name="notification_failed_title">Backup failed</string>
+    <string name="notification_failed_title">Could not make backup.</string>
```
Maybe just "Could not backup"?

comradekingu replied:
That would be "Could not back up", which opens up "Could not back up ___".
It is now ~"A backup could not be made", starting with "Could not".
 "Could not perform backup" is what I would use, but it seems chiefly British.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -177,26 +180,26 @@
     <string name="restore_finished_button">Finish</string>
 
     <string name="restore_storage_skip">Skip restoring files</string>
     <string name="restore_storage_choose_snapshot">Choose a storage backup to restore (experimental)</string>
```
What exactly is a storage backup?
Full partition backup?

chirayudesai replied:
No, you get to choose which part of the (Internal) storage you want to backup.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -160,9 +163,9 @@
     <string name="restore_invalid_location_title">No backups found</string>
-    <string name="restore_invalid_location_message">We could not find any backups at this location.\n\nPlease choose another location that contains a %s folder.</string>
+    <string name="restore_invalid_location_message">Could not find any backups at this location.\n\nPlease choose another location that contains a %s folder.</string>
     <string name="restore_set_error">An error occurred while loading the backups.</string>
```
```suggestion
    <string name="restore_set_error">Could not load the backups.</string>
```

chirayudesai commented:
For example, I think one thing that could go on right away is

Disable/Enable -> Turn off / turn on, depending on context / somebody else double checking.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +97,40 @@
-    <string name="recovery_code_recreated">New recovery code has been created successfully</string>
+    <string name="recovery_code_verification_new_dialog_message">Do you want to generating a new code, making your existing backups inaccessible? They will be deleted if possible.</string>
+    <string name="recovery_code_recreated">New recovery code created</string>
     <string name="recovery_code_auth_title">Re-enter your screen lock</string>
```
What does this mean?

chirayudesai replied:
Copied from AOSP Settings, I think. re-enter whatever you use on the screen lock / lock screen.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +97,40 @@
     <string name="recovery_code_verification_new_dialog_title">Wait one second…</string>
-    <string name="recovery_code_verification_new_dialog_message">Generating a new code will make your existing backups inaccessible. We\'ll try to delete them if possible.\n\nAre you sure you want to do this?</string>
-    <string name="recovery_code_recreated">New recovery code has been created successfully</string>
+    <string name="recovery_code_verification_new_dialog_message">Do you want to generating a new code, making your existing backups inaccessible? They will be deleted if possible.</string>
```
```suggestion
    <string name="recovery_code_verification_new_dialog_message">Generate a new recovery code? (This makes your existing backups inaccessible, and deletes them if possible.)</string>
```

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -94,40 +97,40 @@
     <string name="recovery_code_verification_error_message">You have entered an invalid recovery code. Please try again!\n\nIf you have lost your code, tap on \'Generate new code\' below.</string>
     <string name="recovery_code_verification_try_again">Try again</string>
     <string name="recovery_code_verification_generate_new">Generate new code</string>
     <string name="recovery_code_verification_new_dialog_title">Wait one second…</string>
```
```suggestion
    <string name="recovery_code_verification_new_dialog_title">Invalidating current backups</string>
```
This isn't good, but the figure of speech ~"Hold on there…" isn't carried over.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -68,17 +71,17 @@
     <string name="storage_check_fragment_restore_title">Looking for backups…</string>
-    <string name="storage_check_fragment_backup_error">An error occurred while accessing the backup location.</string>
+    <string name="storage_check_fragment_backup_error">Could not access the backup location.</string>
     <string name="storage_check_fragment_permission_error">Unable to get the permission to write to the backup location.</string>
```
```suggestion
    <string name="storage_check_fragment_permission_error">Grant the app permission to write to the backup location first.</string>
```

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,50 +3,53 @@
     <string name="app_name" translatable="false">Seedvault</string>
     <string name="backup">Backup</string>
```
As in?

chirayudesai replied:
This is just a simple string used in many places (such as the app name on CalyxOS), and thus I don't think it needs any more explanation.

A backup app being called "Backup" is as simple as it gets.

comradekingu replied:
Until it appears for translation with no context, which is hard to read out of being presented with a verb when it should have been a noun or vice versa when translated.

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,50 +3,53 @@
-    <string name="settings_info">All backups are encrypted on your phone. To restore from backup you will need your 12-word recovery code.</string>
-    <string name="settings_auto_restore_title">Automatic restore</string>
+    <string name="settings_info">All backups are encrypted on your device. Restore backups using your 12-word recovery code.</string>
+    <string name="settings_auto_restore_title">Auto-restore</string>
```
```suggestion
    <string name="settings_auto_restore_title">Restoration</string>
```

comradekingu committed:
Applied suggestions

chirayudesai reviewed (commented):
@comradekingu do you have a device running SeedVault?


We definitely cannot take such a big PR, but are happy to take bits and pieces as mentioned above. With proper credits / attribution, of course.

chirayudesai commented:
@comradekingu also if you're on matrix, we're at `#seedvault-app:matrix.org`

comradekingu committed:
Merge branch 'android13' into patch-2

comradekingu committed:
Applied suggestions

comradekingu commented on app/src/main/res/values/strings.xml:
```diff
@@ -180,29 +182,29 @@
     <string name="restore_finished_button">Finish</string>
 
     <string name="restore_storage_skip">Skip restoring files</string>
     <string name="restore_storage_choose_snapshot">Choose a storage backup to restore (experimental)</string>
```
```suggestion
    <string name="restore_storage_choose_snapshot">Choose a former backup of part of the internal storage to restore (experimental)</string>
```

chirayudesai commented:
Given the number of changes still required here, and the fact that something this big will always run into friction.

We do want to improve the strings here, and we'll try to do it step by step to make it easier to manage all around.

Thanks!

comradekingu commented:
@chirayudesai I think being given the work with no end in sight in two years doesn't bode well for the coming decades, but maybe I will come back to see it done someday.

chirayudesai commented:
@comradekingu I agree, but I did mention this a day after the PR got opened - https://github.com/seedvault-app/seedvault/pull/404#issuecomment-1122639062

Such big changes are hard to work with for us, and we can't just entirely re-do everything all at once.

There's so many strings I'd like to change, but it has to happen gradually, a few / related sections at a time, rather than all at once like this.

An example: https://github.com/seedvault-app/seedvault/pull/623

I even have two comments there about other possible changes I'd have liked to make, but I intentionally avoided doing it all at once - that just makes it hard to review and decide, and thus get merged.
